src/MICore/Transports/LocalLinuxTransport.cs

```diff
@@ -187,9 +195,29 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
             reader = new StreamReader(gdbStdOutStream);
         }
 
+        private void FifoWatcher_Deleted(object sender, FileSystemEventArgs e)
+        {
+            if (e.FullPath == _gdbStdInName || e.FullPath == _gdbStdOutName)
+            {
+                Logger?.WriteLine("Gdb fifo deleted, stop debugging");
+                _fifoWatcher.Deleted -= FifoWatcher_Deleted;
+                this.Callback.OnDebuggerProcessExit(null);
+            }
+        }
+
         protected override string GetThreadName()
         {
             return "MI.LocalLinuxTransport";
         }
+
+        public override void Close()
+        {
+            base.Close();
```

Comment:
But then isn’t there still a race condition?

From: Fangliang Xue [mailto:notifications@github.com]
Sent: Wednesday, March 23, 2016 11:24 AM
To: Microsoft/MIEngine <MIEngine@noreply.github.com>
Cc: Jackson Davis <Jackson.Davis@microsoft.com>
Subject: Re: [MIEngine] Stop debugging when in or out fifo is deleted (#239)


In src/MICore/Transports/LocalLinuxTransport.cs<https://github.com/Microsoft/MIEngine/pull/239#discussion_r57211314>:

>          protected override string GetThreadName()

>          {

>              return "MI.LocalLinuxTransport";

>          }

> +

> +        public override void Close()

> +        {

> +            base.Close();

Jackson's question seems disappear after I rebase the branch, the question is "Since the fifos are always deleted on debugging stop, does this now mean we're trigging exit twice?"
I added the code to remove file watcher on Close(), or normal exit Close will be called before the fifo is actually deleted.

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/MIEngine/pull/239/files/06287a0d7c6739fa999ec4a742acf56171654b33#r57211314>